app-frontend/src/app/components/pages/project/layer/scenes/scenes/index.js

```diff
@@ -1,11 +1,201 @@
+            map.setLayer(mapLayerName, mapLayer, true);
+        });
+    }
+
+    removeMapLayers() {
+        return this.getMap().then(map => {
+            map.deleteLayers(mapLayerName);
+        });
+    }
+
+
+    fetchPage(page = this.$state.params.page || 1, filter, order) {
+        // TODO do we need to list ingesting scenes? that stuff goes under filter?
+        // this.getIngestingSceneCount();
+        delete this.fetchError;
+        this.sceneList = [];
+        const currentQuery = this.projectService.getProjectLayerScenes(
+            this.projectId,
+            this.layerId,
+            {
+                pageSize: this.projectService.scenePageSize,
+                page: page - 1
+            }
+        ).then((paginatedResponse) => {
+            this.sceneList = paginatedResponse.results;
+            this.sceneActions = new Map(this.sceneList.map(this.addSceneActions.bind(this)));
+            this.pagination = this.paginationService.buildPagination(paginatedResponse);
+            this.paginationService.updatePageParam(page);
+            if (this.currentQuery === currentQuery) {
+                delete this.fetchError;
+            }
+        }, (e) => {
+            if (this.currentQuery === currentQuery) {
+                this.fetchError = e;
+            }
+        }).finally(() => {
+            if (this.currentQuery === currentQuery) {
+                delete this.currentQuery;
+            }
+        });
+        this.currentQuery = currentQuery;
+        return currentQuery;
+    }
+
+    hasDownloadPermission(scene) {
+        if (this.RasterFoundryRepository.getScenePermissions(scene).includes('download')) {
+            return true;
+        }
+        return false;
+    }
+
+    addSceneActions(scene) {
+        // details, view layers, hide (unapprove), remove (delete from layer)
+        let actions = [{
+            name: 'Remove',
+            title: 'Remove image from layer',
+            callback: () => this.removeScenes([scene]),
+            menu: true
+        }];
+
+        if (this.hasDownloadPermission(scene)) {
+            actions.unshift({
+                icon: 'icon-download',
+                name: 'Download',
+                title: 'Download raw image data',
+                callback: () => this.modalService.open({
+                    component: 'rfSceneDownloadModal',
+                    resolve: {
+                        scene: () => scene
+                    }
+                }).result.catch(() => {}),
+                menu: false
+            });
+        }
+
+        return [scene.id, actions];
+    }
+
+    openImportModal() {
+        const activeModal = this.modalService.open({
+            component: 'rfSceneImportModal',
+            resolve: {
+                project: () => this.project,
+                origin: () => 'project'
+            }
+        });
+
+        activeModal.result.then(results => {
+            this.checkPendingImports();
```

Comment:
I think this looks like an unimplemented function.

Reply:
er yup. I copied that over cus I thought it was something I'd need, but it's not in the designs. I'll delete it.